lnrpc/switchrpc/switch_server.go

```diff
@@ -188,3 +206,250 @@ func (r *ServerShell) CreateSubServer(
 
 	return subServer, macPermissions, nil
 }
+
+// SendOnion handles the incoming request to send a payment using a
+// preconstructed onion blob provided by the caller.
+func (s *Server) SendOnion(_ context.Context,
+	req *SendOnionRequest) (*SendOnionResponse, error) {
+
+	if len(req.OnionBlob) == 0 {
+		return nil, status.Error(codes.InvalidArgument,
+			"onion blob is required")
+	}
+	if len(req.OnionBlob) != lnwire.OnionPacketSize {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"onion blob size=%d does not match expected %d bytes",
+			len(req.OnionBlob), lnwire.OnionPacketSize)
+	}
+
+	if len(req.PaymentHash) == 0 {
+		return nil, status.Error(codes.InvalidArgument,
+			"payment hash is required")
+	}
+
+	amount := lnwire.MilliSatoshi(req.Amount)
+	if amount <= 0 {
```

Comment:
you need to do the check before casting to uint64. If done after casting, it is too late

Reply:
Updated to perform the check before casting.